buildSrc/src/main/groovy/airbyte-python-docker.gradle

```diff
@@ -5,30 +5,25 @@ import org.gradle.api.tasks.Exec
 /*
 This plugin exists to build & test a python module via Gradle without requiring the developer to install Python on their machine. It achieves this by mounting the module into a Python Docker image and running the tests. 
 
-Modules consuming this plugin must define two scripts `run_tests.sh` and `run_format.sh`. 
+Modules consuming this plugin must define two scripts `run_tests.sh`.
 */
 
-class AirbytePythonDockerConfiguration {
-    String moduleDirectory
-}
-
 class AirbytePythonDockerPlugin implements Plugin<Project> {
 
     void apply(Project project) {
-        def extension = project.extensions.create('airbytePythonDocker', AirbytePythonDockerConfiguration)
 
         project.task('airbytePythonDockerApply', type: Exec) {
             /*
             Install the dependencies and run the tests from a docker container
              */
-            commandLine "docker", "run", "-v", "${project.projectDir.getAbsolutePath()}:/home", "--entrypoint", "/bin/bash", "python:3.9-slim", "-c", "chmod +x /home/run_tests.sh && /home/run_tests.sh /home"
-        }
 
-        project.task('blackFormat', type: Exec) {
-            /*
-            Run formatter and static analysis from a docker container
-             */
-            commandLine "docker", "run", "-v", "${project.projectDir.getAbsolutePath()}:/home", "--entrypoint", "/bin/bash", "python:3.9-slim", "-c", "chmod +x /home/run_tests.sh && /home/run_format.sh /home"
+            // This doesn't take into account dependencies if you don't have a lock file. We strongly recommend using a
+            // lock file in other to ensure that given no file changes and two "pip install", you have the same output
+            inputs.dir project.projectDir.getAbsolutePath()
+            outputs.cacheIf { true }
```

Comment:
To be confirmed if needed

Reply:
Based on https://docs.gradle.com/enterprise/tutorials/caching/#introduction_build_avoidance, it seems like it could still be useful as "build avoidance" even though there are no outputs. If locally or remotely a cache entry has the same input, it would avoid us having to re-run the tests